contrib/aws/internal/sns/sns.go

```diff
@@ -0,0 +1,105 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package sns
+
+import (
+	"encoding/json"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/aws/aws-sdk-go-v2/service/sns/types"
+	"github.com/aws/smithy-go/middleware"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+const (
+	datadogKey           = "_datadog"
+	maxMessageAttributes = 10
+)
+
+func EnrichOperation(span tracer.Span, in middleware.InitializeInput, operation string) {
+	switch operation {
+	case "Publish":
+		handlePublish(span, in)
+	case "PublishBatch":
+		handlePublishBatch(span, in)
+	}
+}
+
+func handlePublish(span tracer.Span, in middleware.InitializeInput) {
+	params, ok := in.Parameters.(*sns.PublishInput)
+	if !ok {
+		log.Debug("Unable to read PublishInput params")
+		return
+	}
+
+	traceContext, err := getTraceContext(span)
+	if err != nil {
+		log.Debug("Unable to get trace context: %s", err.Error())
+		return
+	}
```

Comment:
nit.

I think you could simplify this out even further. Since both `handlePublish` and `handlePublishBatch` call `getTraceContext`, you could make the `getTraceContext` call just once, outside of the handle functions and then pass the context objects into the handle functions. 

This makes no functional change, but it would DRY up the code a bit. That would be an improvement, since if we need to update the `getTraceContext` call in the future, we'd right now need to update it in two locations. Making the `getTraceContext` call just once would mean updating code in a single location.

Reply:
Thanks, this is an easy fix and I'm open to doing it. But I'm trying to structure the `EnrichOperation` func in a way that allows other teams to modify other operations in SQS/SNS. Could there be operations where they won't need the trace context? If so, I think it would make more sense to keep the `getTraceContext()` calls in the handle funcs. What do you think?